pkg/action/action_test.go

```diff
@@ -144,6 +144,32 @@ func TestOpFromClaim_UndefinedParams(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestOpFromClaim_ParamsDoesNotApplyToAction(t *testing.T) {
+	now := time.Now()
+	b := mockBundle()
+	// Add a parameter only defined for the test action
+	b.Parameters["param_test"] = bundle.ParameterDefinition{
+		ApplyTo: []string{"test"},
+	}
+	c := &claim.Claim{
+		Created:  now,
+		Modified: now,
+		Name:     "name",
+		Revision: "revision",
+		Bundle:   b,
+		Parameters: map[string]interface{}{
+			"param_one":   "oneval",
+			"param_two":   "twoval",
+			"param_three": "threeval",
+			"param_test":  "only for test action",
+		},
+	}
+	invocImage := c.Bundle.InvocationImages[0]
+
+	_, err := opFromClaim(claim.ActionInstall, c, invocImage, mockSet, os.Stdout)
+	assert.EqualError(t, err, `parameter "param_test" does not apply to action "install"`)
```

Comment:
I think, the param should just be ignored:
if you have an action with "modify:true"  which has an action-specific parameter, this parameter value will be recorded in the claim after execution. If you then try to execute uninstall, you'll get an error `parameter "foo" does not apply to action "uninstall"`

Reply:
Ok right, but said like that it means that the `apply-to` is useless... except for `REQUIRED` parameters specific to an action. I'll update the check with that 👍